image_data_guide/04b_tensorflow_training.ipynb

```diff
@@ -0,0 +1,889 @@
+    "These helper functions define transformations needed to be done to our TFRecords datasets before training. For more in-depth info see the Pre-processing guide in this series."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[34mdef\u001b[39;49;00m \u001b[32mtfrecord_parser\u001b[39;49;00m(record):\n",
+      "    features = {\n",
+      "        \u001b[33m'\u001b[39;49;00m\u001b[33mheight\u001b[39;49;00m\u001b[33m'\u001b[39;49;00m: tf.io.FixedLenFeature([], tf.int64),\n",
+      "        \u001b[33m'\u001b[39;49;00m\u001b[33mwidth\u001b[39;49;00m\u001b[33m'\u001b[39;49;00m: tf.io.FixedLenFeature([], tf.int64),\n",
+      "        \u001b[33m'\u001b[39;49;00m\u001b[33mdepth\u001b[39;49;00m\u001b[33m'\u001b[39;49;00m: tf.io.FixedLenFeature([], tf.int64),\n",
+      "        \u001b[33m'\u001b[39;49;00m\u001b[33mlabel\u001b[39;49;00m\u001b[33m'\u001b[39;49;00m: tf.io.FixedLenFeature([], tf.int64),\n",
+      "        \u001b[33m'\u001b[39;49;00m\u001b[33mimage_raw\u001b[39;49;00m\u001b[33m'\u001b[39;49;00m: tf.io.FixedLenFeature([], tf.string),\n",
+      "    }\n",
+      "    parsed_features = tf.io.parse_single_example(record, features)\n",
+      "    \u001b[34mreturn\u001b[39;49;00m tf.io.decode_jpeg(parsed_features[\u001b[33m'\u001b[39;49;00m\u001b[33mimage_raw\u001b[39;49;00m\u001b[33m'\u001b[39;49;00m]), parsed_features[\u001b[33m'\u001b[39;49;00m\u001b[33mlabel\u001b[39;49;00m\u001b[33m'\u001b[39;49;00m]\n",
+      "\n",
+      "\u001b[34mdef\u001b[39;49;00m \u001b[32maugment\u001b[39;49;00m(image, label):\n",
+      "    image = tf.image.random_flip_left_right(image)\n",
+      "    image = tf.image.random_flip_up_down(image)\n",
+      "    image = tf.image.random_brightness(image, \u001b[34m0.2\u001b[39;49;00m)\n",
+      "    image = tf.image.random_hue(image, \u001b[34m0.1\u001b[39;49;00m)\n",
+      "    \u001b[34mreturn\u001b[39;49;00m (image, label)\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pygmentize \"training_tensorflow/tensorflow_train.py\" | sed -n 7,23p"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<pre>\n",
+    "</pre>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Execution safety\n",
+    "For safety we wrap the training code in this standard if statement though it is not strictly required"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[34mif\u001b[39;49;00m \u001b[31m__name__\u001b[39;49;00m == \u001b[33m\"\u001b[39;49;00m\u001b[33m__main__\u001b[39;49;00m\u001b[33m\"\u001b[39;49;00m:\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pygmentize \"training_tensorflow/tensorflow_train.py\" | sed -n 25p"
```

Comment:
File isn't there.

Reply:
New PR has the training scripts added for pytorch and tensorflow

Reply:
Checked new commit. Works fine.